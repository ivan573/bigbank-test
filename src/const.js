export const Route = {
  HOME: '/',
  GAME: '/game',
  SHOP: '/shop',
  REPUTATION: '/reputation'
}

export const ModalWindowType = {
  QUEST: 'quest',
  QUEST_RESULT: 'questResult',
  PURCHASE_ITEM: 'purchaseItem',
  PURCHASE_RESULT: 'purchaseResult',
  GAME_OVER: 'gameOver'
}
