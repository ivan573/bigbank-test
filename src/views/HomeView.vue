<template>
  <div class="home">
    <GameButton class="home__start-button" text="New game" @click="init" />
  </div>
</template>

<script>
import { Route } from '@/router/index'
import GameButton from '@/components/ui/GameButton.vue'

export default {
  name: 'HomeView',
  components: {
    GameButton
  },
  methods: {
    async init() {
      this.$store.dispatch('endGame')
      await this.$store.dispatch('initGame')
      this.$router.push({ path: Route.GAME })
    }
  }
}
</script>

<style lang="scss" scoped>
  .home {
    &__start-button {
      position: absolute;
      top: 50%;
      left: 50%;
      transform: translate(-50%, -50%);
    }
  }
</style>
