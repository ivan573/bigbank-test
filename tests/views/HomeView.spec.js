import { mount, createLocalVue } from '@vue/test-utils'
import Vuex from 'vuex'
import VueRouter from 'vue-router'
import { Route } from '@/const'
import HomeView from '@/views/HomeView.vue'

describe('HomeView.vue', () => {
  it('matches snapshot', () => {
    const wrapper = mount(HomeView)
    expect(wrapper.element).toMatchSnapshot()
  })

  it('initializes new game on new game button click and redirects to game path', async () => {
    const localVue = createLocalVue()
    localVue.use(Vuex)
    localVue.use(VueRouter)

    const actions = {
      endGame: jest.fn(),
      initGame: jest.fn()
    }

    const store = new Vuex.Store({
      actions
    })
    const router = new VueRouter()

    const wrapper = mount(HomeView, {
      localVue,
      store,
      router
    })

    const newGameButtonElement = wrapper.find('.home__start-button')
    await newGameButtonElement.trigger('click')

    await wrapper.vm.$nextTick()

    expect(actions.endGame).toHaveBeenCalledTimes(1)
    expect(actions.initGame).toHaveBeenCalledTimes(1)
    expect(wrapper.vm.$route.path).toBe(Route.GAME)
  })
})
