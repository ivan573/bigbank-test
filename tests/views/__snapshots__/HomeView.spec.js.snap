// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HomeView.vue matches snapshot 1`] = `
<div
  class="home"
>
  <button
    class="game-button home__start-button"
  >
    <span
      class="game-button__text"
    >
      New game
    </span>
  </button>
</div>
`;
